### DISCUSSION TITLE
How I can save nested Arrays and Autocomplete in a Stepform?

### DISCUSSION THREAD

**Discussion by REDLIGHTFILMS** [react-hook-form/react-hook-form#2044] [Q&A] [answered]
I have the problem that if you add two new task areas and then press Next and go back again, they don't stand up anymore. It is the same with the Autocomplete. What can you do there?

https://codesandbox.io/s/array-problem-evt14

**Comment by bluebill1049** [comment]
you should store your form submitted data in global state/store and then repopulate the form again.

**Reply by REDLIGHTFILMS** [reply]
I already store my data in a global state. How do I fill in the form again?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#useForm search defualtValues

**Comment by REDLIGHTFILMS** [comment, marked as answer]
This is my new code. I added a global state and set it as default. But unfortunately it still doesn't work. The state is being updated. Do you have a solution for this?
https://codesandbox.io/s/array-problem-evt14

**Reply by bluebill1049** [reply]
if state is coming async, use `reset` it's in the doc with an example.

**Reply by REDLIGHTFILMS** [reply]
thx :)
